srcpkgs/openttd/template

```diff
@@ -15,46 +15,67 @@ configure_args="
  -DCMAKE_INSTALL_BINDIR=bin
  -DCMAKE_INSTALL_DATADIR=/usr/share
  -DCMAKE_INSTALL_DOCDIR=share/doc/openttd
+ $(vopt_bool dedicated OPTION_DEDICATED)
 "
 
+build_options="gfx sfx msx dedicated"
+build_options_default="gfx sfx msx"
+desc_option_gfx="Include OpenGFX graphics pack"
+desc_option_sfx="Include OpenSFX sounds pack"
+desc_option_msx="Include OpenMSX music pack"
+desc_option_dedicated="Build dedicated server only, without GUI"
+
 hostmakedepends="pkg-config unzip tar"
-makedepends="SDL2-devel libcurl-devel freetype-devel fontconfig-devel
- harfbuzz-devel icu-devel libpng-devel liblzma-devel fluidsynth-devel"
-depends="hicolor-icon-theme fluidsynth soundfont-fluid"
+makedepends="$(vopt_if dedicated '' 'SDL2-devel freetype-devel fontconfig-devel
+ harfbuzz-devel icu-devel libpng-devel fluidsynth-devel') libcurl-devel
+ liblzma-devel"
+depends="$(vopt_if dedicated '' 'hicolor-icon-theme fluidsynth
+ soundfont-fluid')"
 short_desc="Open Source version of Transport Tycoon Deluxe"
 maintainer="Emil Tomczyk <emru@emru.xyz>"
 license="GPL-2.0-only, Zlib"
 homepage="https://www.openttd.org/"
 distfiles="https://cdn.openttd.org/openttd-releases/${version}/openttd-${version}-source.tar.xz
- https://cdn.openttd.org/opengfx-releases/${_gfxver}/opengfx-${_gfxver}-all.zip
- https://cdn.openttd.org/opensfx-releases/${_sfxver}/opensfx-${_sfxver}-all.zip
- https://cdn.openttd.org/openmsx-releases/${_msxver}/openmsx-${_msxver}-all.zip"
+ $(vopt_if gfx https://cdn.openttd.org/opengfx-releases/${_gfxver}/opengfx-${_gfxver}-all.zip)
+ $(vopt_if sfx https://cdn.openttd.org/opensfx-releases/${_sfxver}/opensfx-${_sfxver}-all.zip)
+ $(vopt_if msx https://cdn.openttd.org/openmsx-releases/${_msxver}/openmsx-${_msxver}-all.zip)"
 checksum="2c14c8f01f44148c4f2c88c169a30abcdb002eb128a92b9adb76baa76b013494
- 928fcf34efd0719a3560cbab6821d71ce686b6315e8825360fba87a7a94d7846
- e0a218b7dd9438e701503b0f84c25a97c1c11b7c2f025323fb19d6db16ef3759
- 5a4277a2e62d87f2952ea5020dc20fb2f6ffafdccf9913fbf35ad45ee30ec762"
+ $(vopt_if gfx 928fcf34efd0719a3560cbab6821d71ce686b6315e8825360fba87a7a94d7846)
+ $(vopt_if sfx e0a218b7dd9438e701503b0f84c25a97c1c11b7c2f025323fb19d6db16ef3759)
+ $(vopt_if msx 5a4277a2e62d87f2952ea5020dc20fb2f6ffafdccf9913fbf35ad45ee30ec762)"
 
 post_extract() {
-	bsdtar xf opengfx-${_gfxver}.tar
-	bsdtar xf opensfx-${_sfxver}.tar
-	bsdtar xf openmsx-${_msxver}.tar
-	rm opengfx-${_gfxver}.tar
-	rm opensfx-${_sfxver}.tar
-	rm openmsx-${_msxver}.tar
+	$(vopt_if gfx "bsdtar xf opengfx-${_gfxver}.tar")
```

Comment:
The `vopt` expansion helpers are for string interpolation. For other control flow, you should be testing the `$build_option_*` variables directly. See other templates that use build options for examples.